ios/FirebaseContinue/FirebaseContinue/Classes/FCNContinue.m

```diff
@@ -16,14 +16,164 @@
+  // of the most recent Activity they may wish to continue within the application.
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^(void) {
+
+    // First, ensure the inputs could be valid.
+    if ([self isNilOrOnlyWhitespace:activityUrl]) {
+      [self
+          executeCompletionBlockOnMainThread:completionBlock
+                                   withError:[self
+                                                 createErrorWithReason:@"activityUrl is invalid"]];
+      return;
+    } else if ([self isNilOrOnlyWhitespace:applicationName]) {
+      [self executeCompletionBlockOnMainThread:completionBlock
+                                     withError:[self createErrorWithReason:
+                                                         @"applicationName is invalid"]];
+      return;
+    }
+
+    // Next, ensure the current user is signed in.
+    FIRUser *currentUser = [FIRAuth auth].currentUser;
+    if (currentUser == nil) {
+      [self executeCompletionBlockOnMainThread:completionBlock
+                                     withError:[self createErrorWithReason:
+                                                         @"The current user must be signed in"]];
+      return;
+    }
+
+    // Get the DatabaseReference for the most recent Activity the current user may wish to continue
+    // (within the context of the "applicationName" application).
+    NSString *databasePathToMostRecentActivity =
+        [NSString stringWithFormat:@"firebaseContinue/%@/%@", applicationName, currentUser.uid];
+    FIRDatabaseReference *mostRecentActivityRef =
+        [[FIRDatabase database] referenceWithPath:databasePathToMostRecentActivity];
+
+    // Since we now have a DatabaseReference for it, delete any current value for the
+    // most recent Activity the user may wish to continue.
+    // We do this because Activities are considered immutable within the database, so
+    // before setting a value any existing value must first be deleted.
+    [mostRecentActivityRef removeValueWithCompletionBlock:^(NSError *_Nullable error,
+                                                            FIRDatabaseReference *_Nonnull ref) {
+      if (error != nil) {
+        [self executeCompletionBlockOnMainThread:completionBlock
+                                       withError:
+                                           [self createErrorWithReason:error.localizedDescription]];
+        return;
+      }
+
+      // Now we can set the new value for the most recent Activity the user may wish
+      // to continue within the application, since any previous value has been deleted.
 
+      // Create a Firebase Continue Activity to use as the new most recent Activity
+      // the user may wish to continue within the application.
+      // The schema of each Activity is defined in
+      // sample-firebase-continue-database.rules.json.
+      NSDictionary *activityMetadata = @{@"addedAt" : [FIRServerValue timestamp]};
```

Comment:
Collections should use lightweight ObjC generics, e.g., NSDictionary<NSString *, NSString *> *.
Since this dictionary has multiple types of objects ([FIRServerValue timestamp] returns another NSDictionary) so you'll have to do NSDictionary<NSString *, id> *, which will allow any ObjC value but limits the keys to NSStrings.